src/parquet/file/writer-internal.cc

```diff
@@ -70,22 +70,21 @@ void SerializedPageWriter::Close(bool has_dictionary, bool fallback) {
   metadata_->WriteTo(sink_);
 }
 
-std::shared_ptr<Buffer> SerializedPageWriter::Compress(
-    const std::shared_ptr<Buffer>& buffer) {
-  // Fast path, no compressor available.
-  if (!compressor_) return buffer;
+void SerializedPageWriter::Compress(const std::shared_ptr<Buffer>& src_buffer,
+    std::shared_ptr<ResizableBuffer>& dest_buffer) {
```

Comment:
We might get into a habit of passing the immutable and mutable args here instead as:

`const Buffer& src_buffer, ResizableBuffer* dest_buffer`